tests/providers/google/cloud/hooks/test_datapipeline.py

```diff
@@ -0,0 +1,128 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import copy
+import re
+import shlex
+from asyncio import Future
+from typing import Any
+from unittest import mock
+from unittest.mock import MagicMock
+from uuid import UUID
+
+import pytest
+
+from airflow.exceptions import AirflowException, AirflowProviderDeprecationWarning
+from airflow.providers.apache.beam.hooks.beam import BeamHook, run_beam_command
+from airflow.providers.google.cloud.hooks.datapipeline import (
+    DataPipelineHook,
+    DEFAULT_DATAPIPELINE_LOCATION
+)
+
+TASK_ID = "test-datapipeline-operators"
+TEST_NAME = "projects/test-datapipeline-operators/locations/test-location"
+TEST_BODY = {
+    "name": "projects/test-datapipeline-operators/locations/test-location/pipelines/test-pipeline",
+            "type": "PIPELINE_TYPE_BATCH",
+            "workload": {
+                "dataflowFlexTemplateRequest": {
+                "launchParameter": {
+                    "containerSpecGcsPath": "gs://dataflow-templates-us-central1/latest/Word_Count_metadata",
+                    "jobName": "test-job",
+                    "environment": {
+                    "tempLocation": "test-temp-location"
+                    },
+                    "parameters": {
+                    "inputFile": "gs://dataflow-samples/shakespeare/kinglear.txt",
+                    "output": "gs://test/output/my_output"
+                    }
+                },
+                "projectId": "test-project-id",
+                "location": "test-location"
+                }
+            }
+}
+TEST_LOCATION = "test-location"
+TEST_PROJECTID = "test-project-id"
+TEST_GCP_CONN_ID = "test_gcp_conn_id"
+TEST_DATA_PIPELINE_NAME = "test_data_pipeline_name"
+TEST_PARENT = "projects/test-datapipeline-operators/locations/test-location/pipelines/test-pipeline"
+
+class TestDataPipelineHook:
+    """
+    Module meant to test the DataPipeline Hooks
+    """
+    
+    def setup_method(self):
+        self.datapipeline_hook = DataPipelineHook(gcp_conn_id = "test_google_cloud_default")
```

Comment:
Is there a reason the conn_id mentioned here `"test_google_cloud_default"` is different than on line 63 `"test_gcp_conn_id"`

Reply:
yes, I was just testing that the test worked without the mock by calling the actual connection before address the mock gcp conn error because I believe it'll take longer than writing the necessary test. once that is fixed, I will update them to match

Reply:
Makes sens

Reply:
Updated to use the actual connection id as suggested since the Dataflow example does the same